### ISSUE TITLE
Issue with DefaultAccountManager.CS

### ISSUE THREAD

**Issue by shimail1** [Azure/azure-batch-samples#113]
Hello Guys

When the new person is downloading the batch explorer code and trying to run first time they are getting exceptions. because the code is trying to de-serialize the xml node and since file is empty created with zero bytes , the code was exploding. 

I manually created the blank account xml file in the app data directory. C:\Users\<username>\AppData\Local\Microsoft\Azure Batch Explorer with following code. then ran again and then the app worked.

`<?xml version="1.0" encoding="UTF-8"?>

-<DefaultAccountSerializationContainer xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">

<Accounts/>

</DefaultAccountSerializationContainer>`

So please fix the code here to not only check if file exists and if it is not zero bytes then de-serialize otherwise write the xml lines first and then de-serialize.

private async Task LoadAccountsAsync()
        {
            string accountFileName = GetAccountFileName();
            if (!File.Exists(accountFileName) OR ALSO CHECK IF FILE EXISTS AS ZERO BYTES )    
            {
                using (var myFile = File.Create(accountFileName))
                {
                }
                await this.SaveAccountsAsync().ConfigureAwait(false);
            }

            //Try first to deserialize using the new account deserialization mechanism, if that fails, fall back to the old way
            bool writeBackToFile = false;
            try
            {
                XmlSerializer x = new XmlSerializer(typeof(DefaultAccountSerializationContainer));
                using (StreamReader reader = new StreamReader(accountFileName))
                {
                    object deserializedObject = x.Deserialize(reader); //TODO: This is not technically async...

                    DefaultAccountSerializationContainer accountContainer = (DefaultAccountSerializationContainer)deserializedObject;

                    List<Account>

**Comment by matthchr** [comment]
Hi @shimail1 - thanks for the report.
I was able to reproduce the issue and fixed the FileLength 0 issue (see: #114), but I can't observe it happening on first BatchExplorer startup.

The old code was: 
```
            if (!File.Exists(accountFileName))
            {
                using (var myFile = File.Create(accountFileName))
                {
                }
                await this.SaveAccountsAsync().ConfigureAwait(false);
            }
```

According to this though, it shouldn't be possible to get an empty accounts.xml if the file doesn't exist - since we do `await this.SaveAccountsAsync().ConfigureAwait(false)` which writes the default content.  Do you by chance know how there got to be an empty accounts.xml file there in the first place?  Possibly from an older version of BatchExplorer, or created manually?

In any case, the issue should be resolved now.

**Comment by shimail1** [comment]
empty accounts.xml file gets created when developer executes the code first time and because the code creates the file in the app data folder but due to exception quits the loop therefore the xml insert does not happen and file remains empty. and next time when the code runs it see the file exists but again generates exception.

**Comment by matthchr** [comment]
This issue resolved by #114
